pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Reader.java

```diff
@@ -116,4 +116,63 @@
      * @return Whether the reader is connected to the broker
      */
     boolean isConnected();
+
+    /**
+     * Reset the subscription associated with this reader to a specific message id.
+     * <p>
+     *
+     * The message id can either be a specific message or represent the first or last messages in the topic.
+     * <p>
+     * <ul>
+     * <li><code>MessageId.earliest</code> : Reset the reader on the earliest message available in the topic
+     * <li><code>MessageId.latest</code> : Reset the reader on the latest message in the topic
+     * </ul>
+     *
+     * Note: this operation can only be done on non-partitioned topics. For these, one can rather perform the seek() on
+     * the individual partitions.
+     *
+     * @param messageId the message id where to reposition the reader
+     */
```

Comment:
shouldn't the operation be renamed as `reset` instead of `seek`. I mean I would assume `seek` to just find the message id but not change the state of subscription.

Reply:
mm I don't agree here, imho keeping `seek()` would be easier for the user understanding because of already existing seek on Consumer.

Also, what I think that is great of naming it `seek()` is that it behaves exactly the same as seeking on a filename e.g. [Linux](http://man7.org/linux/man-pages/man2/lseek.2.html) which a lot of people already know what is meant for.